spk/python38/src/service-setup.sh

```diff
@@ -0,0 +1,17 @@
+PATH="${SYNOPKG_PKGDEST}/bin:$PATH"
+
+service_postinst ()
+{
+    # Install the wheels
+    ${SYNOPKG_PKGDEST}/bin/pip install --no-deps --no-index -U --force-reinstall -f ${SYNOPKG_PKGDEST}/share/wheelhouse ${SYNOPKG_PKGDEST}/share/wheelhouse/*.whl >> ${INST_LOG} 2>&1
+
+    # Log installation informations
+    ${SYNOPKG_PKGDEST}/bin/python3 --version >> ${INST_LOG} 2>&1
+    echo -e "\nSystem installed modules:" >> ${INST_LOG}
+    ${SYNOPKG_PKGDEST}/bin/pip freeze >> ${INST_LOG}
+
+    # Byte-compile in background
+    PYTHON_SHORT_VER=$(${SYNOPKG_PKGDEST}/bin/python3 -c 'import sys; print("{0}.{1}".format(*sys.version_info[:2]))')
+    ${SYNOPKG_PKGDEST}/bin/python3 -m compileall -q -f ${SYNOPKG_PKGDEST}/lib/python${PYTHON_SHORT_VER} >> ${INST_LOG} &
+    ${SYNOPKG_PKGDEST}/bin/python3 -OO -m compileall -q -f ${SYNOPKG_PKGDEST}/lib/python${PYTHON_SHORT_VER} >> ${INST_LOG} &
```

Comment:
@hgy59 I was testing this package on my NAS which is ancient, and I saw that the `compileall` is run twice. 
Do you know why? 
Or maybe @smaarn knows?

Reply:
To be honest this was a brutal copy paste without any questioning on the purpose.
I'll be checking for next package release (a 3.8.8 version was recently released).

Reply:
It seems that both steps are generating different kind of output and are dedicated to generating python compiled files which will be used depending on the "optimization" requested at runtime (see https://docs.python.org/3/using/cmdline.html#cmdoption-o ).
I'm guessing both run could actually be useful, but since I'm no python expert I should rather keep it quiet here ;)

Reply:
The background compilation was always done twice, even in the python 2.7 package of December 2015

``` bash
    # Byte-compile in background
    ${INSTALL_DIR}/bin/python -m compileall -q -f ${INSTALL_DIR}/lib/python2.7 > /dev/null &
    ${INSTALL_DIR}/bin/python -OO -m compileall -q -f ${INSTALL_DIR}/lib/python2.7 > /dev/null &
```

So I suppose this is done by intention.
The first line creates `*.pyc` files and the second one `*.pyo`. But I don't know whether this has any benefit.

Reply:
@Safihre @hgy59 As a matter of fact, if we **really** wanted to keep some kind of logic we should even re-run a third time it with a single `-O` optimization :(.

At this stage I wonder if we shouldn't extract this logics into separate dedicated scripts ? From what I see most projects are actually running python without any optimization specified so only the first command is useful to them.
The second command would only be useful to code being run with `-OO` switch. 

I see the following leads:

* Remove the second run.
  * according to the documentation, only the **initial** loading time of projects using that switch would be impacted
* Remove the second run and add a script which could be run to force the compileall to be run with optimization parameters being provided as arguments
  * Would need being run at install time by projects requiring it (but then it would need to cache the various optimization requirements asked to rerun those upon upgrade)
  * Would need being run at startup time by projects requiring it

WDYT ?

Reply:
In my understanding (not 100% sure), .pyo are "drop-in" replacements for .pyc, with assertions and __debug__ stripped (-O) and additionally docstrings stripped (-OO). Precompiling them reduces first load time for runtimes that make use of them. In order to use a .pyo, the runtime must be started with -O/-OO (please confirm/falsify).

So separate runs for -O and -OO would just overwrite each other, since both generate .pyo.

If compile execution time is the issue here, the -OO run might be removed at potential initial loading time cost for some usages of this python SPK. If compile execution time is not a big deal, leaving things unchanged will be the safe option.

Reply:
Now I understand. We at SABnzbd do run with `-OO` so it is usefull. I didn't know it created seperate files on each run.
Thanks for letting me know :)

Reply:
@Safihre Thanks, I was close to ask for known usages of -OO runtimes in my last reply. But since this is a generic python package, we would never know ;-)

Reply:
Actually, since 3.8 they don't "overlap" each other. Hence the fact that in prior versions only two runs were "necessary". (see https://www.python.org/dev/peps/pep-0488/ for more details)

Reply:
Indeed, the python38 package does not create `*.pyo` files. It creates `*.pyc` and `*.opt-2.pyc` files as by `PEP 488`.

As long as none of the packages runs python with `-O` we don't need to create respective bytecode at installation.

Reply:
Just to prevent future misunderstandings: pep-0488 has been implemented already in python 3.5, not 3.8 (https://bugs.python.org/issue23731). So this is not an issue regarding 3.7->3.8 upgrade.